src/waveform/waveformwidgetfactory.cpp

```diff
@@ -713,6 +713,7 @@ void WaveformWidgetFactory::swap() {
                 }
                 QGLWidget* glw = qobject_cast<QGLWidget*>(pWaveformWidget->getWidget());
                 if (glw != nullptr) {
+                    glw->makeCurrent();
```

Comment:
Does it still work when guarding it with: 
if (QGLContext::currentContext() != context()) {

Reply:
What should be context() in this statement? I didn't get the context here;-)

Reply:
ups, glw->context()

Reply:
This code seems also work without failures: ![grafik](https://user-images.githubusercontent.com/64457745/100163210-80375400-2eb5-11eb-92c0-c24b9b0071b3.png)

Reply:
Combine it with the `if` condition above: `if (glw != nullptr && glw->context() != QGLContext::curentContext()) {`

Reply:
nevermind, that isn't logically equivalent

Reply:
@daschuer does `glw->makeCurrent()` have any performance penalty if the context is already current? Is the check for `glw->context() != QGLContext::currentContext()` needed?

Reply:
The problem seems to be, that the openGL context is sometimes not set to the correct thread: https://doc.qt.io/qt-5/qopenglcontext.html#makeCurrent
In this cases swapBuffers fails, but swapBuffers has to be executed always.

Reply:
> @daschuer does glw->makeCurrent() have any performance penalty if the context is already current? Is the check for glw->context() != QGLContext::currentContext() needed?

makeCurrent() is expensive. We have the check everywhere else for some reasons. 

Maybe this is done internally anyway. in this case we can omit the check. 
I need to debug into the code to be sure.  

The other puzzling question is why it works with real hardware. Maybe we are doing something wrong that steals the context from the thread in the Angle case.

Reply:
> Maybe this is done internally anyway. 

I checked. It is not. We should do the check before calling makeCurrent.
https://code.woboq.org/qt5/qtbase/src/opengl/qgl.cpp.html#_ZN10QGLContext11makeCurrentEv